qiskit/backends/_projectq_simulator.py

```diff
@@ -0,0 +1,256 @@
+                    qubit_index = operation['qubits'][0]
+                    qubit = qureg[qubit_index]
+                    clbit = operation['clbits'][0]
+                    Measure | qubit
+                    bit = 1 << clbit
+                    self._classical_state = (
+                        self._classical_state & (~bit)) | (int(qubit)
+                                                           << clbit)
+                    # check whether we already measured this qubit
+                    if operation['qubits'][0] in unmeasured_qubits:
+                        unmeasured_qubits.remove(operation['qubits'][0])
+                # Check if reset
+                elif operation['name'] == 'reset':
+                    qubit = operation['qubits'][0]
+                    raise SimulatorError('Reset operation not yet implemented '+
+                                         'for ProjectQ C++ backend')
+                elif operation['name'] == 'barrier':
+                    pass
+                else:
+                    backend = self._configuration['name']
+                    err_msg = '{0} encountered unrecognized operation "{1}"'
+                    raise SimulatorError(err_msg.format(backend,
+                                                        operation['name']))
+            for ind in unmeasured_qubits:
+                qubit = qureg[ind]
+                Measure | qubit
+            eng.flush()
+            # Turn classical_state (int) into bit string            
+            state = format(self._classical_state, 'b')
+            outcomes.append(state.zfill(self._number_of_clbits))
+        # Return the results
+        counts = dict(Counter(outcomes))
+        data = {'counts': _format_result(
+            counts, cl_reg_index, cl_reg_nbits)}
+        if self._shots == 1:
+            data['quantum_state'] = self._quantum_state
+            data['classical_state'] = self._classical_state,
+        return {'data': data, 'status': 'DONE'}
+
+def _get_register_specs(bit_labels):
+    """
+    Get the number and size of unique registers from bit_labels list.
+
+    Args:
+        bit_labels (list): this list is of the form::
+
+            [['reg1', 0], ['reg1', 1], ['reg2', 0]]
+
+            which indicates a register named "reg1" of size 2
+            and a register named "reg2" of size 1. This is the
+            format of classic and quantum bit labels in qobj 
+            header.
+
+    Returns:
+        iterator of register_name:size pairs.
+    """
+    it = itertools.groupby(bit_labels, operator.itemgetter(0))
+    for register_name, sub_it in it:
+        yield register_name, max(ind[1] for ind in sub_it) + 1
+
+def _format_result(counts, cl_reg_index, cl_reg_nbits):
+    """Format the result bit string.
+
+    This formats the result bit strings such that spaces are inserted
+    at register divisions.
+
+    Args:
+        counts : dictionary of counts e.g. {'1111': 1000, '0000':5}
```

Comment:
Would it be possible to add a note about the other two arguments to the docstring?

Reply:
yes!